pkg/codegen/dotnet/gen.go

```diff
@@ -398,7 +411,7 @@ func (mod *modContext) typeString(t schema.Type, qualifier string, input, state,
 	case *schema.ResourceType:
 		if strings.HasPrefix(t.Token, "pulumi:providers:") {
 			pkgName := strings.TrimPrefix(t.Token, "pulumi:providers:")
-			return fmt.Sprintf("Pulumi.%s.Provider", namespaceName(mod.namespaces, pkgName))
+			return mod.rootNamespace + "." + namespaceName(mod.namespaces, pkgName) + ".Provider"
```

Comment:
I'll switch it back to`Sprintf`. I'm happy to use either. I generally prefer `+` when most of the string components are behind variables (instead of string literals).